controllers/remote/cluster_cache.go

```diff
@@ -51,21 +51,62 @@ const (
 
 // ClusterCacheTracker manages client caches for workload clusters.
 type ClusterCacheTracker struct {
-	log    logr.Logger
-	client client.Client
-	scheme *runtime.Scheme
+	log                   logr.Logger
+	clientDisableCacheFor []client.Object
+	client                client.Client
+	scheme                *runtime.Scheme
 
 	lock             sync.RWMutex
 	clusterAccessors map[client.ObjectKey]*clusterAccessor
 }
 
+type clusterCacheTrackerOptions struct {
+	log                   logr.Logger
+	clientDisableCacheFor []client.Object
+}
```

Comment:
Could we just export this instead of having variable functions?
```suggestion
type ClusterCacheTrackerOptions struct {
	Logger                   logr.Logger
	ClientDisableCacheFor []client.Object
}
```